ReactAndroid/src/main/java/com/facebook/react/devsupport/DevServerHelper.java

```diff
@@ -218,8 +218,10 @@ public String getInspectorDeviceUrl() {
   /**
    * @return the host to use when connecting to the bundle server from the host itself.
    */
-  private static String getHostForJSProxy() {
-    return AndroidInfoHelpers.DEVICE_LOCALHOST;
+  private String getHostForJSProxy() {
+    String host = getDebugServerHost();
```

Comment:
Fixed the `? :` formatting and pushed a non null assertion (defensive if also possible, if you insist).